s3fs/core.py

```diff
@@ -943,6 +944,10 @@ async def _put_file(self, lpath, rpath, chunksize=50 * 2 ** 20, **kwargs):
             else:
                 await self._mkdir(lpath)
         size = os.path.getsize(lpath)
+
+        content_type = mimetypes.guess_type(lpath)
+        kwargs.update({"ContentType": content_type[0]})
```

Comment:
Wouldn't this override the user provided one?

Reply:
Also `mimetypes.guess_type` can return `None`, so we will be effectively sending a null header if we can't guess. We need to do this conditionally (and through using `setdefault`).

Reply:
good call, i've added in a check. right now im ignoring the encoding, should i add that in aswell ?.

Reply:
You can propose this in a separate PR, but I am thinking the encoding is even less useful than MIME, I would need to be convinced that mimetypes actually does a good job at guessing.